forum/qa-include/pages/message.php

```diff
@@ -1,257 +1,288 @@
+
+    $hideForm = !empty($pageerror) || $messagesent;
+
+    $qa_content['title'] = qa_lang_html('misc/private_message_title');
+
+    $qa_content['error'] = @$pageerror;
+
+    $blockedPrivateMessages = qa_db_query_sub('SELECT `fromUserId`, `toUserId` FROM ^blockedpw WHERE fromUserId = # AND toUserId = # OR fromUserId = # AND toUserId = #', $loginuserid, $toaccount['userid'], $toaccount['userid'], $loginuserid);
+    $allowedPrivateMessages = qa_opt('allow_private_messages');
+    $blockedPrivateMessageBool = 0 != $blockedPrivateMessages->num_rows;
+
+    if (
+        $blockedPrivateMessageBool ||
+        !$allowedPrivateMessages
+    ) {
+        $userLevel = qa_get_logged_in_level();
+        if($userLevel < QA_USER_LEVEL_EDITOR) {
+            // user blocked pw and loggedin can't send messages
+            $qa_content['custom'] = 'Nie możesz wysłać wiadomości prywatnej do tego uzytkownika.';
+        } else {
+            // user blocked pw, but loggedin is admin and can send message
+            $qa_content['form_message'] = [
+                'tags' => 'method="post" action="' . qa_self_html() . '"',
+                'style' => 'tall',
+                'ok' => $messagesent ? qa_lang_html('misc/message_sent') : null,
+                'fields' => [
+                    'message' => [
+                        'type' => $hideForm ? 'static' : '',
+                        'label' => qa_lang_html_sub('misc/message_for_x', qa_get_one_user_html($handle, false)),
+                        'tags' => 'name="message" id="message"',
+                        'value' => qa_html(@$inmessage, $messagesent),
+                        'rows' => 8,
+                        'note' => qa_lang_html_sub('misc/message_explanation', qa_html(qa_opt('site_title'))),
+                        'error' => qa_html(@$errors['message']),
+                    ],
+                ],
+                'buttons' => [
+                    'send' => [
+                        'tags' => 'onclick="qa_show_waiting_after(this, false);"',
+                        'label' => qa_lang_html('main/send_button'),
+                    ],
+                ],
+                'hidden' => [
+                    'domessage' => '1',
+                    'code' => qa_get_form_security_code('message-' . $handle),
+                ],
+            ];
+        }
+    } else {
+		$qa_content['form_message'] = [
+                'tags' => 'method="post" action="' . qa_self_html() . '"',
+                'style' => 'tall',
+                'ok' => $messagesent ? qa_lang_html('misc/message_sent') : null,
+                'fields' => [
+                    'message' => [
+                        'type' => $hideForm ? 'static' : '',
+                        'label' => qa_lang_html_sub('misc/message_for_x', qa_get_one_user_html($handle, false)),
+                        'tags' => 'name="message" id="message"',
+                        'value' => qa_html(@$inmessage, $messagesent),

```

Comment:
@ do wywalenia

Reply:
To jest kod q2a, niezbyt chce go ruszać.

Reply:
do jest dodane przez Ciebie więc nie q2a